### PR TITLE
add delete function to comments with optimistic rendering

### PR DESCRIPTION
**Pull request by gcltukjs2022**

### CODE REVIEW

File: src/Components/Comments.jsx [hunk 1]
```diff
@@ -15,6 +17,24 @@ export default function Comments({ article_id }) {
     });
   }, [comments]);
 
+  const handleDeleteComment = (event, comment_id) => {
+    event.preventDefault();
+    alert("You have deleted a comment!");
```
Comment (Augs0) [inline src/Components/Comments.jsx]: I don't think this is the best way to provide feedback. Could we render something on screen perhaps?

File: src/Components/Comments.jsx [hunk 2]
```diff
@@ -15,6 +17,24 @@ export default function Comments({ article_id }) {
...
+
+    const newCommentsList = comments.filter((comment) => {
+      return comment.comment_id !== comment_id;
+    });
+
+    setComments(newCommentsList);
+
+    removeComment(comment_id)
+      .then(() => {})
```
Comment (Augs0) [inline src/Components/Comments.jsx]: Why is this then block empty?

**Review by Augs0 — CHANGES_REQUESTED** [review]
This is almost there, George and well done again for getting so far! I would think about changing how you let users know they deleted a comment but otherwise I'm happy with this :)

**Review by leekli — APPROVED** [review]
Thanks for making these changes George it looks much better.

You are now free to merge this branch.

**Pull request merged**